lineapy/graph_reader/node_collection.py

```diff
@@ -93,6 +93,9 @@ class NodeCollection:
     sliced_nodes: Set[LineaID] = field(default_factory=set)
     raw_codeblock: str = field(default="")
     is_empty: bool = field(default=True)
+    pre_computed_artifact: Union[None, Tuple[str, Optional[int]]] = field(
```

Comment:
can we pull a boolean out here like 
`is_pre_computed_artifact` that is initialized in the post_init and use this in the if else statement below? 

This way it is obvious that the behavior of node collection is different and it is clear what a None precomputed artifact means.